backend/cinema/views.py

```
@@ -166,71 +219,98 @@ def get_queryset(self):
         movie_id_str = self.request.query_params.get("movie")
 
         queryset = self.queryset
-
         if date:
-            date = datetime.strptime(date, "%Y-%m-%d").date()
-            queryset = queryset.filter(show_time__date=date)
-
+            try:
+                date_obj = datetime.strptime(date, "%Y-%m-%d").date()
```

You parse `date` using `datetime.strptime` and silently ignore ValueError (pass). Consider surfacing invalid input back to the client with a 400 response and an explanatory error message instead of ignoring it; silent ignores can make debugging client-side problems harder. If you intentionally want permissive behavior, document it in the API docs so clients know an invalid date is treated as “no date filter”.